### ISSUE TITLE
 Cannot read property 'serialize' of undefined

### ISSUE THREAD

**Issue by borntodesign** [angular/angular-cli#1900]
Trying to use sass on main global stylesheet:

ng set defaults.styleExt scss

Error:

Cannot read property 'serialize' of undefined
TypeError: Cannot read property 'serialize' of undefined
    at /Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:235:87
    at JsonSerializer.property (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/serializer.ts:100:9)
    at _loop_1 (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:235:32)
    at /Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:240:17
    at JsonSerializer.object (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/serializer.ts:88:9)
    at ObjectSchemaTreeNode.serialize (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:227:20)
    at /Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:231:92
    at JsonSerializer.property (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/serializer.ts:100:9)
    at _loop_1 (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:231:36)
    at /Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:240:17
    at JsonSerializer.object (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/serializer.ts:88:9)
    at ObjectSchemaTreeNode.serialize (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:227:20)
    at /Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:293:33
    at JsonSerializer.array (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/serializer.ts:107:9)
    at ArraySchemaTreeNode.serialize (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:291:20)
    at /Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:231:92
    at JsonSerializer.property (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/serializer.ts:100:9)
    at _loop_1 (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:231:36)
    at /Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:240:17
    at JsonSerializer.object (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/serializer.ts:88:9)
    at RootSchemaTreeNode.ObjectSchemaTreeNode.serialize (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-tree.ts:227:20)
    at GeneratedSchemaClass.SchemaClassBase.$$serialize (/Users/lorne/projects/hsi-magazine/node_modules/angular-cli/addon/ng2/models/json-schema/schema-class-factory.ts:141:25)

**Comment by borntodesign** [comment]
Well I have been using angular cli for about two months working on a large
app with lots of different libraries installed. So my problem could be to
do with upgrading as well. I did follow the guide :)

**Comment by neojack647** [comment]
I received the same exact error with 'ng set defaults.styleExt less' after upgrading my app this morning.
Error:
TypeError: Cannot read property 'serialize' of undefined
    at C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:235:87
    at JsonSerializer.property (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\serializer.ts:100:9)
    at _loop_1 (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:235:32)
    at C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:240:17
    at JsonSerializer.object (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\serializer.ts:88:9)
    at ObjectSchemaTreeNode.serialize (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:227:20)
    at C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:231:92
    at JsonSerializer.property (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\serializer.ts:100:9)
    at _loop_1 (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:231:36)
    at C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:240:17
    at JsonSerializer.object (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\serializer.ts:88:9)
    at ObjectSchemaTreeNode.serialize (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:227:20)
    at C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:293:33
    at JsonSerializer.array (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\serializer.ts:107:9)
    at ArraySchemaTreeNode.serialize (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:291:20)
    at C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:231:92
    at JsonSerializer.property (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\serializer.ts:100:9)
    at _loop_1 (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:231:36)
    at C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:240:17
    at JsonSerializer.object (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\serializer.ts:88:9)
    at RootSchemaTreeNode.ObjectSchemaTreeNode.serialize (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-tree.ts:227:20)
    at GeneratedSchemaClass.SchemaClassBase.$$serialize (C:\git\aws.clients\oms-allocation-testing\node_modules\angular-cli\addon\ng2\models\json-schema\schema-class-factory.ts:141:25)

**Comment by neojack647** [comment]
angular-cli@1.0.0-beta.11-webpack.8 is what was installed with the update.

**Comment by neojack647** [comment]
$ ng --version
angular-cli: 1.0.0-beta.11-webpack.8
node: 5.7.1
os: win32 ia32
$ node --version
v5.7.1

**Comment by neojack647** [comment]
Same issue as #1905

**Comment by borntodesign** [comment]
I did get it to work manually by setting the default style extension to scss in angular-cli.json. I also had to restart the server.

**Comment by tchoulihan** [comment]
What's that line look like in `angular-cli.json`, to use scss?

**Comment by johnkorzhuk** [comment]
"defaults": {
    "styleExt": "css" }

**Comment by intellix** [comment]
SCSS was introduced as the new "main syntax" in version 3, six years ago: https://github.com/sass/sass/releases/tag/3.0.0

I feel like the `.sass` extension should be deprecated :P Are people still using it?

**Comment by omargon** [comment]
@borntodesign I follow your steps manually setting the `styleExt` to `scss`, but if I rename the main `style.css` to `style.scss` I get the following error: 
```
ERROR in multi styles
Module not found: Error: Can't resolve '/src/styles.css' in '/node_modules/angular-cli/models'
 @ multi styles
```

Did you have to change anything else?

**Comment by RomanTheOne** [comment]
@omargon I believe you have to change the styles within your angular-cli.json to:
`"styles": [
        "styles.scss"
      ],`

**Comment by borntodesign** [comment]
Of course make sure you restart your server as well ;) something I did not
do at first lol

**Comment by filipesilva** [comment]
To be clear, the real but here is that `ng set defaults.styleExt scss` doesn't work, but should. We'll fix it.

**Comment by skovmand** [comment]
Same thing applies to ```ng set defaults.styleExt less```

**Comment by RicardoVaranda** [comment]
The issue is in the ng set command so anything that comes after more than likely will give you an error, give it time as Filipe Silva said it's already on the Angular reams list of things to fix :)

**Comment by lirantal** [comment]
Can we re-open this as it is still reproducing with `angular-cli: 1.0.0-beta.17`
cc @filipesilva @borntodesign

**Comment by RicardoVaranda** [comment]
@lirantal This fix was only merged in the recent 1.0.0-beta.18:

See
https://github.com/angular/angular-cli/blob/master/CHANGELOG.md#100-beta18-2016-10-20

**Comment by lirantal** [comment]
Got it, thanks for clearing this up!
